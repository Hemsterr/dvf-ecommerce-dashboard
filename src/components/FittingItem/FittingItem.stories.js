// Libs
import React from 'react'
import { storiesOf } from '@storybook/react'

// Components
import FittingItem from './index'

storiesOf('FittingItem', module).add('Default', () => (
  <FittingItem title="I would like to fit myself using Hemster's Fitting Kit" />
))
