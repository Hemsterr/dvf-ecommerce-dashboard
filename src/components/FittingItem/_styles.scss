.fitting-item {

  &__icon {
    width: pxToRem(75px);
    height: pxToRem(75px);
    display: block;
    background: url('../../assets/icons/fitting-icon.png') no-repeat;
    margin: 0 auto;
  }

  &__appointment {
    width: pxToRem(130px);

    & span {
      background-position: 0 pxToRem(-75px);
    }
  }

  &__fitting {
    width: pxToRem(200px);

    & span {
      background-position: 0 pxToRem(-225px);
    }
  }

  &__ship {
    width: pxToRem(144px);
  }

  &__deliver {
    width: pxToRem(170px);

    & span {
      background-position: 0 pxToRem(-150px);
    }
  }

  &__title, &__description {
    color: $black;
  }

  &__title {
    @include text-size(pxToRem(12px), pxToRem(1px), pxToRem(30px));
    text-transform: capitalize;
  }

  &__description {
    @include text-size(pxToRem(10px), pxToRem(1px), pxToRem(20px));
  }
}


@media only screen and (max-width: $screen-md) {
  .fitting-item  {
    display: flex;
    justify-content: center;
    width: auto;
    padding-bottom: pxToRem(32px);

    &__wrapper {
      max-width: pxToRem(176px);
      padding-left: pxToRem(32px);
    }
  }
}

@media only screen and (max-width: $screen-tn) {
  .fitting-item   {
    &__icon {
      width: pxToRem(60px);
      height: pxToRem(60px);
      background-size: cover;
    }

    &__wrapper {
      padding-left: pxToRem(15px);
    }

    &__appointment {
      & span {
        background-position: 0 pxToRem(-60px);
      }
    }

    &__fitting {
      & span {
        background-position: 0 pxToRem(-181px);
      }
    }

    &__deliver {
      & span {
        background-position: 0 pxToRem(-119px);
      }
    }
  }
}
