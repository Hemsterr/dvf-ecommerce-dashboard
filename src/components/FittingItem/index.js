// @flow
import React from 'react'

type Props = {
  title: string,
  description: string,
}

class FittingItem extends React.PureComponent<Props> {
  static defaultProps = {
    title: '',
    description: '',
  }

  render() {
    const { description, title } = this.props
    return (
      <div className={`fitting-item fitting-item__${title}`}>
        <div>
          <span className="fitting-item__icon" />
        </div>
        <div className="fitting-item__wrapper">
          <p className="fitting-item__title">{title}</p>
          <p className="fitting-item__description">{description}</p>
        </div>
      </div>
    )
  }
}

export default FittingItem
