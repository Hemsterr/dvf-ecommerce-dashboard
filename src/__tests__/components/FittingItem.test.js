// Libs
import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

// Component
import FittingItem from '../../components/FittingItem'

const mockProps = {
  description: 'Book an appointment to meet with our Fit Stylist virtually',
  title: 'appointment',
}

const wrapper = (props = {}) => {
  const component = shallow(<FittingItem {...props} />)

  return component
}

describe('FittingItem component', () => {
  // Snapshot
  test('should render correctly FittingItem component default props', () => {
    const component = wrapper()
    expect(shallowToJson(component)).toMatchSnapshot()
    expect(FittingItem.defaultProps.description).toEqual('')
    expect(FittingItem.defaultProps.type).toEqual('')
  })

  // Testing props
  test('should be rendered with props', () => {
    const wrapperWithProps = wrapper(mockProps)
    expect(
      wrapperWithProps.find(`.fitting-item__${mockProps.title}`).length
    ).toEqual(1)
    expect(
      wrapperWithProps.find('.fitting-item__title').props().children
    ).toEqual(mockProps.title)
    expect(
      wrapperWithProps.find('.fitting-item__description').props().children
    ).toEqual(mockProps.description)
  })
})
