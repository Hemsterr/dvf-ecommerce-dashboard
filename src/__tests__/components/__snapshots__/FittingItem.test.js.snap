// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FittingItem component should render correctly FittingItem component default props 1`] = `
<div
  className="fitting-item fitting-item__appointment"
>
  <div>
    <span
      className="fitting-item__icon"
    />
  </div>
  <div
    className="fitting-item__wrapper"
  >
    <p
      className="fitting-item__title"
    >
      appointment
    </p>
    <p
      className="fitting-item__description"
    >
      Book an appointment to meet with our Fit Stylist virtually
    </p>
  </div>
</div>
`;
